### DISCUSSION TITLE
Not specifically Vercel related but I want to run a script once a day that pulls in current data from a multipage API, joins it together then uploads it as 'data.JSON' to my github, triggering a redeployment. How would one do this?

### DISCUSSION THREAD

**Discussion by PaulHaze** [vercel/vercel#6654] [Help]
So as the title says, this isn't specifically a Vercel issue, but I am hoping some of the bright minds on here might be able to help a beginner out with this problem.

I am building a site that at the moment gets its data from a Wordpress/Gravity Forms api. The api is limited to a max of 100 results per page, and eventually there will be 6-7 pages. I need ALL the results to build the data table and charts that the site needs. The front end is being built in react.

So currently the way it is working is that each visit by a user triggers these 6-7 api calls. Then the app builds the front end from the received data. However the data will rarely be changing, so a much better option would be to have a script that once a day makes the calls to the api, builds a file with the returned data, uploads that to the repo the site is being built from and then have my site/repo reference that static file.

Thing is I am still a bit new to this, so whilst I am sure this is possible, I am not really sure how to go about this. If someone here could give me some pointers on where to start that would be great.

Alternatively if someone could build the script for me and get it all set up I would be happy to pay them a fair price for their time.

Thanks.....

**Comment by TooTallNate** [comment]
This sounds like a perfect use-case for [Incremental Static Regeneration](https://vercel.com/docs/next.js/incremental-static-regeneration), which would allow you to (re)build those dynamic pages on-demand based on a specified revalidation threshold. Hope that helps!
